xds/src/main/java/io/grpc/xds/GcpAuthenticationFilter.java

```diff
@@ -119,17 +128,23 @@ public ClientInterceptor buildClientInterceptor(FilterConfig config,
       public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
           MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
 
-        /*String clusterName = callOptions.getOption(XdsAttributes.ATTR_CLUSTER_NAME);
-        if (clusterName == null) {
+        String clusterName = callOptions.getOption(CLUSTER_SELECTION_KEY);
+        if (clusterName.startsWith("cluster_specifier_plugin:")) {
           return next.newCall(method, callOptions);
-        }*/
+        }
 
-        // TODO: Fetch the CDS resource for the cluster.
-        // If the CDS resource is not available, fail the RPC with Status.UNAVAILABLE.
+        XdsConfig xdsConfig = callOptions.getOption(XDS_CONFIG_CALL_OPTION_KEY);
+        StatusOr<XdsClusterConfig> xdsCluster = xdsConfig.getClusters().get(clusterName);
+        CdsUpdate cdsUpdate = xdsCluster.getValue().getClusterResource();
+        if (cdsUpdate == null) {
+          // fail the RPC with Status.UNAVAILABLE.
+          return new FailingClientCall<>(Status.UNAVAILABLE.withDescription("CDS resource unavailable"));
+        }
 
-        // TODO: Extract the audience from the CDS resource metadata.
-        // If the audience is not found or is in the wrong format, fail the RPC.
-        String audience = "TEST_AUDIENCE";
+        if (!cdsUpdate.filterMetadata().containsKey(filterInstanceName)) {
+          return next.newCall(method, callOptions);
+        }
+        String audience = String.valueOf(cdsUpdate.filterMetadata().get(filterInstanceName));
```

Comment:
Don't coerce unexpected types to String. gRFC A83:
> If a cluster metadata entry exists for the filter's instance name, but the entry is of a type other than extensions.filters.http.gcp_authn.v3.Audience, then the filter will fail data plane RPCs with status UNAVAILABLE.

Reply:
If the type is not `String`, it must have been from a different typeUrl/JSON, so it should then fail data plane RPCs.

Although that does allow a simple JSON string to work as well. We should probably create a wrapper (`static class Audience { final String audience; }`)  and return that instead from the parser, just so we know the result came from our parser.